### PR TITLE
Hetzner: add TLSA support and fix RRset write paths

### PR DESCRIPTION
**Pull request by jWeck-Software**
This adds TLSA support to the Hetzner provider.

Hetzner’s RRSet API accepts TLSA values in the regular zone-file style, so the provider now sends TLSA records as:

```text
<usage> <selector> <matching> <hex data>
```

and parses the same format back from `list_rrset`.

While testing this against the live API, two existing RRSet write-path problems showed up:

`add_to_rrset` was sending `ttl` in the `add_records` request body. Hetzner rejects that when the RRSet already exists with a different TTL:

```text
has differing record TTLs
```

The fix is to send only the records to `add_records`, then call `change_ttl` afterwards.

The other issue was the fallback create path. Instead of falling back to `POST /zones/{zone}/rrsets`, the provider now stays on the RRSet action endpoints and uses `set_records` / `add_records` plus `change_ttl`. This avoids the uniqueness edge case that was discussed in #50.

The tests cover TLSA serialization/parsing, multi-value TLSA RRSets, relative Hetzner-style names such as `_25._tcp.mail2`, and the updated add/set fallback behavior.

Validation done:

```text
cargo test hetzner
```

Also tested manually against the live Hetzner API using a real API token and zone. The manual checks covered TLSA set/list/add/remove/delete, appending to an existing RRSet with a different TTL, and creating a fresh RRSet without using `POST /rrsets`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "dns-update"
 description = "Dynamic DNS update (RFC 2136 and cloud) library for Rust"
-version = "0.5.4"
```
Comment (mdecimus) [inline Cargo.toml:4]: Do not modify `.gitignore`, `Cargo.toml` or `CHANGELOG`.

Comment (jWeck-Software) [reply Cargo.toml:4]: Ok, no problem. I have reverted these changes. 1d7ee7c2c5a8a6becbd917846360355b27533d59

**Comment by jWeck-Software** [comment]
Reverted every change to `.gitignore`, `Cargo.toml` and `CHANGELOG`.
-> https://github.com/stalwartlabs/dns-update/pull/76/changes

**Comment by mdecimus** [comment]
Thanks, after your commit I noticed some inefficient AI generated code so I moved the common functions to `utils.rs`, please use those instead rather than re-implementing.
